components/service/glean/README.md

```diff
@@ -71,11 +71,16 @@ excellent place to perform this operation is within the `onCreate` method of the
 Android's `Application` class.
 
 ```Kotlin
+import org.mozilla.yourApplication.GleanMetrics.Pings
+
 class SampleApplication : Application() {
 
     override fun onCreate() {
         super.onCreate()
 
+        // OPTIONAL: Initialize your application's custom pings, if it has any.
+        @Suppress("UNUSED_VARIABLE") val _pings = Pings
```

Comment:
I don't think this is an acceptable requirement.

What if, instead, we generated a `PingRegistry` at build time, instead of populating it at run-time when we hit the constructor?

Reply:
I haven't been able to find any way to make this happen, given the requirement of having multiple `pings.yaml` files (for the app, libraries and glean itself).  You could generate a ping registry for *one* of those files, but you would still have to do something like this to merge them into a central registry.  The problem is that Kotlin doesn't have non-lazy static initialization, as far as I can tell.  (It's not a bad design choice given how problematic those things are in C++ for example).

Reply:
Just brainstorming here, would it be possible to have the parser write a file-class-function that contained these `init` lines, that we could then call from `Glean.init()`?

Reply:
A workable alternative (that removes the need for the @Suppress) might be:

`Glean.registerCustomPings(Pings)`

(and this method itself would be a no-op ;)

Reply:
I haven't really looked at how the parser is handling this yet, so I may be on a tangent here.

Reply:
> Just brainstorming here, would it be possible to have the parser write a file-class-function that contained these init lines, that we could then call from Glean.init()?

No, since this is generated code, `Glean.initialize()` doesn't know about it.

Maybe you could stick the package name in a `buildConfig` variable and use introspection to get at it.  But (a) introspection is problematic and (b) that wouldn't work for libraries...

Reply:
> I haven't really looked at how the parser is handling this yet, so I may be on a tangent here.

Doesn't matter -- we could extend the generator to generate whatever we need.  I'm still not sure what that buys us since we couldn't run that code implicitly, without doing *something* like this.

Reply:
@mdboom Is this a problem only for the debug activity?

Reply:
Looks like we could try to use *object expressions* in this case to work around the problem, [see here](https://kotlinlang.org/docs/tutorials/kotlin-for-py/objects-and-companion-objects.html#object-expressions).